tests/manual/media/FXMediaPlayer/src/fxmediaplayer/media/FXMedia.java

```diff
@@ -0,0 +1,157 @@
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.io.IOException;
+import java.util.stream.Stream;
+import java.nio.file.Files;
+import java.util.function.Predicate;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.nio.file.NoSuchFileException;
+
+/**
+ * <p>Helper class to generate playlist from embeded media files located under
+ * src/fxmediaplayer/media. This is useful to test "file", "jar" and "jrt" protocol.
+ * Copy supported media files under src/fxmediaplayer/media,
+ * compile "FXMediaPlayer", run it and go to "Play List" tab and you should see
+ * your embeded media files.</p><p>
+ *
+ * FILE protocol:
+ * <br>
+ * cd rt/tests/manual/media/FXMediaPlayer
+ * <br>
+ * and run
+ * </p><p>
+ *
+ * JAR protocol:
+ * <br>
+ * cd rt/tests/manual/media/FXMediaPlayer
+ * <br>
+ * ant
+ * <br>
+ * java @../../../../build/run.args -jar dist/FXMediaPlayer.jar
+ * </p><p>
+ *
+ * JRT protocol:
+ * <br>
+ * cd rt/tests/manual/media/FXMediaPlayer
+ * <br>
+ * ant
+ * <br><br>
+ * [macOS/Linux]<br>
+ * jlink --output dist/FXMediaPlayer -p ../../../../build/jmods:dist --add-modules javafx.controls,javafx.media,FXMediaPlayer --launcher FXMediaPlayer=FXMediaPlayer/fxmediaplayer.FXMediaPlayer
+ * <br>
+ * ./dist/FXMediaPlayer/bin/FXMediaPlayer
+ * <br><br>
+ * [Windows]<br>
+ * jlink --output dist/FXMediaPlayer -p ../../../../build/jmods;dist --add-modules javafx.controls,javafx.media,FXMediaPlayer --launcher FXMediaPlayer=FXMediaPlayer/fxmediaplayer.FXMediaPlayer
+ * <br>
+ * dist\FXMediaPlayer\bin\FXMediaPlayer.bat
+ * </p>
+ */
+public class FXMedia {
+
+    private static List<String> SUPPORTED_EXT = new ArrayList<>();
+    private static boolean isJRT = true;
+
+    static {
+        SUPPORTED_EXT.add("mp3");
+        SUPPORTED_EXT.add("wav");
+        SUPPORTED_EXT.add("aif");
+        SUPPORTED_EXT.add("mp4");
+        SUPPORTED_EXT.add("m4a");
+        SUPPORTED_EXT.add("m4v");
+    }
+
+    public static List<String> getEmbededMediaFiles() {
+        List<String> sources = new ArrayList<>();
+
+        try {
+            Path path = null;
+            URI uri = null;
+            Stream<Path> walk = null;
+            try {
+                FileSystem fileSystem = FileSystems.getFileSystem(URI.create("jrt:/"));
+                path = fileSystem.getPath("modules", "FXMediaPlayer",
+                        "fxmediaplayer", "media");
+                walk = Files.walk(path, 1);
+            } catch (NoSuchFileException ex) {
+                isJRT = false;
+                uri = FXMedia.class.getResource("/fxmediaplayer/media").toURI();
+
+                if (uri.getScheme().equals("jar")) {
+                    FileSystem fileSystem = FileSystems.newFileSystem(uri,
+                            Collections.<String, Object>emptyMap());
+                    path = fileSystem.getPath("fxmediaplayer", "media");
+                } else {
+                    path = Path.of(uri);
```

Comment:
should we explicitly check if it is a file?
if("file".equals(uri.getScheme()) { ...

Reply:
I do not see a point of doing it, since it should be a file.